packages/story-editor/src/components/library/panes/shared/hooks/useNestedRovingTabIndex.js

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from '@googleforcreators/react';
+import { useKeyDownEffect } from '@googleforcreators/design-system';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../app';
+
+const SELECTOR = '[tabIndex="-1"]:not(:disabled)';
+
+// @todo Combine this with `useRovingTabIndex` so that if depth is set, this file is used, otherwise the default one is used.
+// @todo Move common helpers into a shared file between the two.
+
+/**
+ * A point in 2D space.
+ *
+ * @typedef {Object} Point2D A point in 2D space.
+ * @property {number} x The X coordinate.
+ * @property {number} y The Y coordinate.
+ */
+
+/**
+ * Returns the center of a given element.
+ *
+ * @param {Element} e The element
+ * @return {Point2D} The coordinates of the center as defined by
+ * getBoundingClientRect's `top` and `left` fields.
+ */
+function getCenter(e) {
+  const rect = e.getBoundingClientRect();
+  return {
+    x: rect.left + rect.width / 2,
+    y: rect.top + rect.height / 2,
+  };
+}
+
+/**
+ * Calculates the square of the distance between 2 points.
+ *
+ * @param {Point2D} p1 The first point.
+ * @param {Point2D} p2 The second point.
+ * @return {number} The square of the distance between the 2 points.
+ */
+function getDistanceSq(p1, p2) {
+  return Math.pow(p1.x - p2.x, 2) + Math.pow(p1.y - p2.y, 2);
+}
+
+/**
+ * Returns either `previousSibling` or `nextSibling` depending on whether the
+ * document is RTL and the user pressed the left or right key.
+ *
+ * @param {boolean} isRTL Whether the document is RTL.
+ * @param {string} key The key pressed.
+ * @return {string} Either `prev` or `next`.
+ */
+export function getFocusableElementDirection(isRTL, key) {
+  const isPreviousDirection = isRTL
+    ? key === 'ArrowRight' || key === 'ArrowUp' || key === 'PageUp'
+    : key === 'ArrowLeft' || key === 'ArrowUp' || key === 'PageUp';
+  return isPreviousDirection ? 'previousSibling' : 'nextSibling';
+}
+
+/**
+ * Gets the parent element of the active element based on given depth.
+ *
+ * @param e
+ * @param depth
+ * @returns {*}
+ */
+function getParentByDepth(e, depth) {
```

Comment:
This gets the parent element that is on the same level with the parent elements of other buttons. Perhaps `getAncestorByDepth` would be more accurate name here 🤔